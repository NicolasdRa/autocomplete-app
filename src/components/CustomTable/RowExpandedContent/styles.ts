import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme: Theme) =>
	createStyles({
		row: {
			align: 'left',
		},

		firstCell: {
			width: '6rem',
			borderBottom: 'none',
		},

		secondCell: {
			display: 'flex',
			flexDirection: 'column',
			borderBottom: 'none',
		},

		title: {
			paddingBottom: '.5rem',
		},

		img: {
			borderRadius: '.3rem',
			width: '10rem',
			margin: '.5rem',
			'&:first-child': { marginLeft: 0 },
		},
	}),
);
