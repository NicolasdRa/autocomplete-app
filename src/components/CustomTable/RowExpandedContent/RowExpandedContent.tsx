import { Box, TableRow, TableCell, Typography } from '@material-ui/core';

import { useStyles } from './styles';

interface Props {
	data: string;
}

export const RowExpandedContent: React.FC<Props> = ({ data }) => {
	const classes = useStyles();

	const firstCellContent = data.split(',').slice(0, 1).join();

	console.log(firstCellContent);

	return (
		<TableRow className={classes.row}>
			<TableCell className={classes.firstCell}></TableCell>
			<TableCell className={classes.secondCell}>
				<Box>
					<Typography variant='subtitle2' className={classes.title}>
						Photo Gallery
					</Typography>
				</Box>
				<Box>
					{data.split(',').map((item) => (
						<img
							loading='lazy'
							className={classes.img}
							src={item}
							alt='thumbnail'
						/>
					))}
				</Box>
			</TableCell>
		</TableRow>
	);
};
