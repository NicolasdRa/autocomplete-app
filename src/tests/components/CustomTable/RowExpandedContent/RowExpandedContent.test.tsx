import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { RowExpandedContent } from '../../../../components/CustomTable/RowExpandedContent/RowExpandedContent';

const mockData = '1, 2, 3';

describe('tests on <RowExpandedContent/>', () => {
	const wrapper = shallow(<RowExpandedContent data={mockData} />);

	test('should render <RowExpandedContent/> correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should render as many images as items divided by commas in the string', () => {
		expect(wrapper.find('img').length).toBe(3);
	});
});
