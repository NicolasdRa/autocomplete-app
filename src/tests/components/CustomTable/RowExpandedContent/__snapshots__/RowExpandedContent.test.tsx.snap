// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <RowExpandedContent/> should render <RowExpandedContent/> correctly 1`] = `
<WithStyles(ForwardRef(Table))>
  <WithStyles(ForwardRef(TableBody))>
    <WithStyles(ForwardRef(TableRow))
      className="makeStyles-row-1"
    >
      <Hidden
        smDown={true}
      >
        <WithStyles(ForwardRef(TableCell))
          className="makeStyles-firstCell-2"
        />
      </Hidden>
      <WithStyles(ForwardRef(TableCell))
        className="makeStyles-secondCell-3"
      >
        <Styled(MuiBox)>
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-title-4"
            variant="subtitle2"
          >
            Photo Gallery
          </WithStyles(ForwardRef(Typography))>
        </Styled(MuiBox)>
        <Styled(MuiBox)>
          <img
            alt="thumbnail"
            className="makeStyles-img-5"
            key="0"
            loading="lazy"
            src="1"
          />
          <img
            alt="thumbnail"
            className="makeStyles-img-5"
            key="1"
            loading="lazy"
            src=" 2"
          />
          <img
            alt="thumbnail"
            className="makeStyles-img-5"
            key="2"
            loading="lazy"
            src=" 3"
          />
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(TableCell))>
    </WithStyles(ForwardRef(TableRow))>
  </WithStyles(ForwardRef(TableBody))>
</WithStyles(ForwardRef(Table))>
`;
